### DISCUSSION TITLE
Script with multiple files

### DISCUSSION THREAD

**Discussion by thesynman** [pybricks/support#289] [Q&A]
Hi,
How do I use pybricksdev to run a script that is split across multiple files? Sorry if that should be obvious but I'm new to python and I don't fully understand how module imports work!
Thanks

**Comment by dlech** [comment]
The currently supported Powered Up hubs don't have file systems (due to limited flash memory) and therefore can't support this. You could try concatenating the files into a single file. https://stackoverflow.com/questions/13613336/python-concatenate-text-files

**Reply by thesynman** [reply]
FYI I have written an extension to pybricksdev that pre-processes the script file and inlines imports. I'll create a pull request soonish.

**Reply by thesynman** [reply]
Pull request now open: https://github.com/pybricks/pybricksdev/pull/20

**Reply by laurensvalk** [reply]
Thanks @thesynman. If you're interested, we also have an experimental implementation that makes some firmware changes to support multi-file projects in https://github.com/pybricks/pybricks-micropython/pull/115. We'd love to get your feedback and ideas for this feature.
